<?xml version="1.0" ?>
<launch>
  <arg name="frame_id" default="world" />
  <arg name="voxel_size" default="0.4" />
  <arg name="voxels_per_side" default="16" />

  <node name="voxblox_skeletonizer" pkg="voxblox_skeleton" type="skeletonizer" clear_params="true" args="-v=2 -minloglevel=2" output="screen" respawn="true">
    <param name="generate_by_layer_neighbors" value="false" />
    <!-- If using full euclidean: 0.78 (45 degrees) -->
    <!-- If using quasi-Euclidean: 1.57 (90 degrees) -->
    <param name="min_separation_angle" value="0.78" />
    <param name="update_esdf" value="false" />
    <param name="min_gvd_distance" value="0.5" />
    <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
    <param name="tsdf_voxels_per_side" value="$(arg voxels_per_side)" />
    <param name="esdf_max_distance_m" value="5.0" />
    <param name="esdf_min_diff_m" value="0.0" />
    <param name="esdf_add_occupied_crust" value="true" />
    <param name="world_frame" value="$(arg frame_id)" />
    <param name="frame_id" value="$(arg frame_id)" />
    <param name="verbose" value="false" />
    
    
    <!-- Publishing settings. -->
    <param name="publish_tsdf_map" value="false" />
    <param name="publish_esdf_map" value="false" />
    <param name="publish_slices" value="false" />
    <param name="publish_tsdf_info" value="false" />
    <param name="publish_pointclouds" value="false" />
    <param name="output_mesh_as_pointcloud" value="false" />
    <param name="output_mesh_as_pcl_mesh" value="false" />
    <param name="publish_traversable" value="false" />
    <param name="traversability_radius" value="1.0" />
    <param name="use_const_weight" value="true" />

  </node>

</launch>